File: frontend/src/types/blackjack.ts
import { BaseGameState, BasePlayer, Card } from './game'

export type HandStatus = 'playing' | 'stand' | 'bust' | 'blackjack' | 'double'

export interface BlackjackHand {
  cards: Card[]
  bet: number
  status: HandStatus
}

export interface BlackjackPlayer extends BasePlayer {
  hands: BlackjackHand[]
}

export interface DealerHand {
  cards: Card[]
}

export interface BlackjackGameState extends BaseGameState {
  gameType: 'blackjack'
  currentPlayerIndex: number
  players: BlackjackPlayer[]
  dealerHand: DealerHand
}

File: backend/internal/models/blackjack.go
package models

type BlackjackHand struct {
	Cards  []Card `json:"cards"`
	Bet    int64  `json:"bet"`
	Status string `json:"status"` // playing, stand, bust, blackjack, double
}

type BlackjackPlayer struct {
	Player
	Hands []BlackjackHand `json:"hands"`
}

type DealerHand struct {
	Cards []Card `json:"cards"`
}

type BlackjackGameState struct {
	BaseGameState
	CurrentPlayerIndex int               `json:"currentPlayerIndex"`
	Players            []BlackjackPlayer `json:"players"`
	DealerHand         DealerHand        `json:"dealerHand"`
}
